### PR TITLE
[Chore](refactor) remove some unused code

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
remove some unused code

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.24 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17162254269 Bytes

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.34 seconds
 stream load tsv:          537 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162302675 Bytes

File: be/src/pipeline/pipeline_task.cpp [hunk 1]
```diff
@@ -284,7 +284,7 @@ Status PipelineTask::finalize() {
 }
 
 Status PipelineTask::try_close() {
-    _sink->try_close(_state);
+    RETURN_IF_ERROR(_sink->try_close(_state));
```
Comment (HappenLee) [inline be/src/pipeline/pipeline_task.cpp:287]: sink try close error， still need to try close source

File: be/src/exprs/runtime_filter.cpp [hunk 2]
```diff
@@ -1899,8 +1894,20 @@ Status RuntimePredicateWrapper::get_push_exprs(std::vector<vectorized::VExprSPtr
...
+
+        TTypeDesc type_desc = create_type_desc(PrimitiveType::TYPE_BOOLEAN);
+        type_desc.__set_is_nullable(false);
+        TExprNode node;
+        node.__set_type(type_desc);
+        node.__set_node_type(TExprNodeType::COMPOUND_PRED);
+        node.__set_opcode(TExprOpcode::COMPOUND_AND);
+        node.__set_is_nullable(min_pred->is_nullable() || max_pred->is_nullable());
+
+        auto and_pred = vectorized::VCompoundPred::create_shared(node);
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter.cpp:1906]: change to and expr，maybe recheck the expr still can push down to storage layer？

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.29 seconds
 stream load tsv:          508 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17168253644 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.92 seconds
 stream load tsv:          505 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161275613 Bytes

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.9 seconds
 stream load tsv:          507 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.0 seconds inserted 10000000 Rows, about 344K ops/s
 storage size: 17160306610 Bytes

File: be/src/exprs/runtime_filter.h [hunk 3]
```diff
@@ -368,8 +368,8 @@ class IRuntimeFilter {
     // expr index
     int _expr_order;
     // used for await or signal
-    doris::Mutex _inner_mutex;
```
Comment (yiguolei) [inline be/src/exprs/runtime_filter.h:371]: 这种的别改，否则不知道是系统的还是我们的了

File: be/src/exprs/runtime_filter.cpp [hunk 4]
```diff
@@ -1477,7 +1476,7 @@ void IRuntimeFilter::init_profile(RuntimeProfile* parent_profile) {
         }
         DCHECK(parent_profile != nullptr);
         _name = fmt::format("RuntimeFilter: (id = {}, type = {})", _filter_id,
-                            ::doris::to_string(_runtime_filter_type));
```
Comment (yiguolei) [inline be/src/exprs/runtime_filter.cpp:1480]: 这种的不要动，能感受到是全局的，否则容易乱

Comment (yiguolei) [reply be/src/exprs/runtime_filter.cpp:1480]: 我们应该全局的函数调用，都加namespace 才对，否则区分不清楚是不是对象的函数

File: be/src/exprs/runtime_filter.cpp [hunk 5]
```diff
@@ -1490,8 +1489,7 @@ void IRuntimeFilter::init_profile(RuntimeProfile* parent_profile) {
 
 void IRuntimeFilter::update_runtime_filter_type_to_profile() {
     if (_profile != nullptr) {
-        _profile->add_info_string("RealRuntimeFilterType",
-                                  ::doris::to_string(_wrapper->get_real_type()));
```
Comment (yiguolei) [inline be/src/exprs/runtime_filter.cpp:1494]: and here

File: be/src/vec/exprs/vexpr.h [hunk 6]
```diff
@@ -81,8 +81,6 @@ class VExpr {
     VExpr() = default;
     virtual ~VExpr() = default;
 
-    virtual VExprSPtr clone() const = 0;
```
Comment (yiguolei) [inline be/src/vec/exprs/vexpr.h:84]: This method is useless????

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.39 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17167290751 Bytes

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**